src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -61,6 +94,7 @@ public sealed partial class NpgsqlReadBuffer
         internal NpgsqlReadBuffer(
             NpgsqlConnector connector,
             Stream stream,
+            Socket socket,
```

Comment:
This should be nullable - we actually pass null from PregeneratedMessages, and check for null inside.